website/pages/docs/k8s/connect/ingress-gateways.mdx

```diff
@@ -5,4 +5,174 @@ sidebar_title: Ingress Gateways
+When deploying the helm chart you must provide helm with a custom yaml file that contains your environment configuration.
+The following yaml snippet is the launching point for a valid configuration that must be supplied when installing using the [official consul-helm chart](https://hub.helm.sh/charts/hashicorp/consul).
+Information on additional options can be found [here](https://www.consul.io/docs/k8s/helm). Configuration options for ingress gateways reside under the [ingressGateways](https://www.consul.io/docs/k8s/helm#v-ingressgateways) entry.
+
+```shell-session
+$ cat > config.yaml <<EOF
+connectInject:
+  enabled: true
+ingressGateways:
+  enabled: true
+  gateways:
+    - name: ingress-gateway
+      replicas: 2
+      service:
+       type: LoadBalancer
+EOF
+```
+
+The gateways stanza is where you will define and configure the set of ingress gateways you want deployed to your environment.
+The only required field for each entry is `name`, though entries may contain any of the fields found in the `defaults` stanza.
+Values in this section override the values from the defaults stanza for the given ingress gateway with one exception.
+The annotations from the defaults stanza will be appended to any user-defined annotations defined in the gateways stanza.
+Please refer to the ingress gateway configuration [documentation](https://www.consul.io/docs/k8s/helm#v-ingressgateways-defaults) for a detailed explanation of each option.
+
+-> *Note*: Make sure any ports that will be used as listeners in the ingress gateway's Consul config entry are included in the `ports` object for each gateway.
+
+## Deploying the helm chart
+
+Ensure you have the latest consul-helm chart by running the following command:
+
+```shell-session
+ $ helm repo add hashicorp https://helm.releases.hashicorp.com
+```
+
+Assuming you saved your configuration to a file called config.yaml in the local directory, you can apply your configuration with the following command:
+
+```shell-session
+$ helm install -f config.yaml <consul-release> hashicorp/consul --wait
+```
+
+## Registering a service with Consul
+
+Now that Kubernetes has been updated, you must add the corresponding configuration to Consul. This requires you to use the consul cli.
+Registering a service with Consul is a multi-step process that consists of:
+
+* Accessing the Consul agent
+* Submitting an Ingress Gateway configuration entry to Consul
+
+### Accessing the Consul agent
+
+You can access the Consul server directly from your host via `kubectl port-forward`, this is helpful for interacting with the consul-ui locally as well as validating connectivity of the application.
+
+```shell-session
+$ kubectl port-foward <consul-release>-server-0 8500 &
+```
+
+-> Be sure the latest consul binary is installed locally on your host.
+[https://releases.hashicorp.com/consul/1.8.0/](https://releases.hashicorp.com/consul/1.8.0/)
+
+-> If TLS and ACLs are enabled, you must set the following environment variables.
```

Comment:
TLS and ACLs are separate concerns, this needs to be split into: If TLS is enabled, set `export CONSUL_HTTP_ADDR=https://localhost:8501`, `export CONSUL_HTTP_SSL_VERIFY=false`

If ACLs are enabled, set `export CONSUL_HTTP_TOKEN=$(kubectl get secret consul-bootstrap-acl-token -o jsonpath={.data.token} | base64 -D)`

☝️ is a good example of why setting global.name is so important, this one liner can just be copy pasted.